### PR TITLE
divide code into components

### PR DESCRIPTION
**Pull request by NormalDuck**
There are 3 new components:
`ScoringInput`: A wrapper around `NumberInput` but you can add text
`CheckboxText`: Checkbox but you can add text to describe what the checkbox does
`PhaseToggle`: A component that takes a array and builds a button group.
`ScoringSection`: Divides `ScoringInput` into sections

### CODE REVIEW

File: app/auto/page.tsx [hunk 1]
```diff
@@ -21,111 +24,96 @@ export default function Home() {
...
           />
-          <text>Low</text>
-          <NumberInput
-            defaultValue={localDraft.auto.highSpecimen}
+          <ScoringInput
+            defaultValue={localDraft.auto.highNet}
             onChange={(val) => {
               setLocalDraft({
                 ...localDraft,
-                auto: { ...localDraft.auto, highSpecimen: val },
```
Comment (Jxwoon) [inline app/auto/page.tsx:91]: Missing the specimen logic

**Review by Jxwoon — CHANGES_REQUESTED** [review]

File: app/auto/page.tsx (hunk 1 above)
Comment (NormalDuck) [reply app/auto/page.tsx:91]: fixed it

File: app/end/page.tsx [hunk 2]
```diff
@@ -17,79 +20,70 @@ export default function Home() {
...
+      />
 
       <div className="dark:text-gray-100">
-        <text className="flex h-10 justify-center">Scored Sample</text>
-        <div className="grid grid-flow-col grid-rows-2 justify-center gap-5 text-center">
-          <NumberInput
-            defaultValue={localDraft.end.highNet}
+        <ScoringSection sectionName="Scored Sample">
+          <ScoringInput
+            defaultValue={localDraft.end.net}
```
Comment (Jxwoon) [inline app/end/page.tsx:34]: NIT: fix scheme

**Review by Jxwoon — CHANGES_REQUESTED** [review]

**Comment by NormalDuck** [comment]
I have made changes to the most of the schema and added some new components. The visuals should match the actual matches. I didn't remove `final` page since I might need it later.

**Review by johnylamw — CHANGES_REQUESTED** [review]
AscentMap needs to modified to also include a "none" state. Right now, the options are either "Observation" or "Level x". There should be an option for when a robot does neither of these.

Additionally, refactor all instances of "Acent" to "Ascent" for proper spelling since it may confuse people working on the code. There might be a select all option in your code editor to refactor all instances of it rather than needing you to do it by hand.

**Comment by johnylamw** [comment]
Also, a fun idea would be to create a VERTICAL slider for ascent states. Higher on the slider means the robot reached a higher climb. But that will require you to add labels to the axis. Maybe add some color coding to designate the level they're at. 

(just a fun idea if you're bored. I think the drop-downs that you have are more intuitive though and simpler to manage imo)

**Comment by NormalDuck** [comment]
I have fixed **all** the typos with cspell. Added a new component in the newest commit and refactored the code to use the new component called `SelectInputText` Added none for ascent.

**Review by johnylamw — CHANGES_REQUESTED** [review]
"Observation" option missing for end-game and there is now 2 options for level 3. 
(small typos like these are natural - that's why pull requests exist)

There's also a "test" SelectInputText on the End pages. General practice would be to comment out or remove any sort of test interfaces considering that you're pushing into main which will be used in production. Personally I don't have any problems with that but just keep that in mind when you start working/interning in the industry. You can keep it there if you plan on further testing with it.

Overall, I like the new changes w/ the components and good job on fixing the spelling mistakes.

**Comment by NormalDuck** [comment]
I didn't look carefully when I pushed the code😅

**Review by johnylamw — APPROVED** [review]
Good on my end. 
@Jxwoon thoughts?

File: app/end/page.tsx [hunk 3]
```diff
@@ -1,164 +1,191 @@
...
+            });
+          }}
+          description="Ascent"
+        >
+          <option>None</option>
+          <option>Observation</option>
+          <option>Level1</option>
+          <option>Level2</option>
+          <option>Level3</option>
+          <option>Level3</option>
```
Comment (Jxwoon) [inline app/end/page.tsx:68]: Nit: why is there two level3?

**Review by Jxwoon — APPROVED** [review]
Don't want to hold up this PR. For the most part this looks good to go! Just left one nit comment that could be addressed in another PR or this one

File: app/end/page.tsx (hunk 3 above)
Comment (NormalDuck) [reply app/end/page.tsx:68]: I didn't look carefully my bad

**Pull request merged**